src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.service.impl;
+
+import java.util.Collections;
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user).get();
+        shoppingCart.getTickets().add(ticketDao.add(new Ticket(movieSession, user)));
+    }
+
+    @Override
+    public Optional<ShoppingCart> getByUser(User user) {
+        return shoppingCartDao.getByUser(user);

```

Comment:
```suggestion
        return shoppingCartDao.getByUser(user);
```

add getting value from optional or throw exception